apps/web/graphql/withApollo.tsx

```diff
@@ -1,19 +1,24 @@
-import React from 'react'
+import React, { useEffect, useState } from 'react'
+import { useRouter } from 'next/router'
 import { NextPageContext } from 'next'
 import { ApolloProvider } from 'react-apollo'
+import { getLocaleFromPath } from '@island.is/web/i18n/withLocale'
 import initApollo from './client'
 
 export const withApollo = (Component) => {
   const NewComponent = ({ apolloState, pageProps }) => {
+    const { asPath } = useRouter()
+    const clientLocale = getLocaleFromPath(asPath)
     return (
-      <ApolloProvider client={initApollo(apolloState)}>
+      <ApolloProvider client={initApollo({ ...apolloState, clientLocale })}>
```

Comment:
Maybe clientLocale should be a separate paramater to initApollo, rather than add data to apolloState unrelated to apolloState?